### PR TITLE
basic pagination component

### PR DESCRIPTION
**Pull request by oskar-kupski-elpassion**
#17 Please let me know if that is what you had in mind

### CODE REVIEW

File: src/components/PPagination.vue [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+
+<script>
+export default {
+  name: 'PPagination',
+  props: {
+    value: Number,
+    totalPages: Number
+  },
+  data() {
+    const blockClassName = 'puto-pagination'
```
Comment (lubien) [inline src/components/PPagination.vue:46]: If this never changes, put directly on template

File: src/components/PPagination.vue [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+    const itemBaseClassName = `${blockClassName}__item`
```
Comment (lubien) [inline src/components/PPagination.vue:47]: `puto-pagination-item`

File: src/components/PPagination.vue [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+    return {
+      blockClassName,
+      linkClassName: `${blockClassName}__link`,
```
Comment (lubien) [inline src/components/PPagination.vue:50]: `puto-pagination-link`

File: src/components/PPagination.vue [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+      itemBaseClassName,
+      prevLabel: '<',
```
Comment (lubien) [inline src/components/PPagination.vue:52]: This and nextLabel would be nice to be a prop.

Also don't abbreviate, use `previousLabel` please

File: src/components/PPagination.vue [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+      nextLabel: '>',
+      firstPage: 1
```
Comment (lubien) [inline src/components/PPagination.vue:54]: Weird. Why this?

File: src/components/PPagination.vue [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+    }
+  },
+  methods: {
+    changePage(page) {
+      this.$emit('input', page)
+    },
+    itemClassName(page) {
+      return [
+        this.itemBaseClassName,
+        this.value === page && `${this.itemBaseClassName}--active`
```
Comment (lubien) [inline src/components/PPagination.vue:80]: `puto-pagination-item-active`

File: src/components/PPagination.vue [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+        `${this.itemBaseClassName}--next`,
+        this.value >= this.totalPages && `${this.itemBaseClassName}--disabled`
+      ]
+    }
+  },
+  methods: {
+    changePage(page) {
+      this.$emit('input', page)
+    },
+    itemClassName(page) {
```
Comment (lubien) [inline src/components/PPagination.vue:77]: I think this is short enough to be directly on the template using `class=""` for static classes and `:classes="{ 'puto-pagination-item-active': CONDITION }`

File: src/components/PPagination.vue [hunk 8]
```diff
@@ -0,0 +1,87 @@
+<template>
+  <ul :class="blockClassName">
+    <li :class="prevItemClassName" v-if="totalPages > firstPage">
+      <span :class="linkClassName" v-if="value <= firstPage">{{ prevLabel }}</span>
+      <a
+        href="#"
+        :class="linkClassName"
+        @click.prevent="changePage(value - 1)"
```
Comment (lubien) [inline src/components/PPagination.vue:8]: Would changePage(-1) be possible to call?

**Comment by lubien** [comment]
image

Oh I'm sorry. I've reviewd but didnt send it!

**Review by lubien — CHANGES_REQUESTED** [review]

**Comment by oskar-kupski-elpassion** [comment]
Hi @lubien, unfortunately I won't be able to help anymore. Please unassign me

**Comment by lubien** [comment]
> Hi @lubien, unfortunately I won't be able to help anymore. Please unassign me

No problem. Best regards. Hope to work with you again if possible.

**Pull request closed without merging**